### PR TITLE
RUN: Don't use `-Cpanic=abort` when `Run with Coverage`

### PR DESCRIPTION
**Pull request by mchernyavsky**
Fixes https://github.com/intellij-rust/intellij-rust/issues/5973.

### CODE REVIEW

File: coverage/src/main/kotlin/org/rust/coverage/GrcovRunner.kt [hunk 1]
```diff
@@ -86,8 +86,7 @@ class GrcovRunner : RsDefaultProgramRunnerBase() {
             val environmentVariables = EnvironmentVariablesData.create(
                 oldVariables.envs + mapOf(
                     "CARGO_INCREMENTAL" to "0",
-                    "RUSTFLAGS" to "-Zprofile -Ccodegen-units=1 -Copt-level=0 -Clink-dead-code -Coverflow-checks=off -Zpanic_abort_tests -Cpanic=abort",
-                    "RUSTDOCFLAGS" to "-Cpanic=abort"
+                    "RUSTFLAGS" to "-Zprofile -Ccodegen-units=1 -Copt-level=0 -Clink-dead-code -Coverflow-checks=off"
```
Comment (Undin) [inline coverage/src/main/kotlin/org/rust/coverage/GrcovRunner.kt:89]: How does it affect users except fix proc-macro lib tests?

Comment (mchernyavsky) [reply coverage/src/main/kotlin/org/rust/coverage/GrcovRunner.kt:89]: Yes, now user can catch panic. Also, previously we incorrectly showed coverage in some cases:

* before:
 

* after:

Comment (Undin) [reply coverage/src/main/kotlin/org/rust/coverage/GrcovRunner.kt:89]: Thanks!

**Review by Undin — APPROVED** [review]
bors r+

**Pull request merged**